InfiniTAM/Engine/RosPoseSourceEngine.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright 2014-2015 Isis Innovation Limited and the authors of InfiniTAM
+
+#include "RosPoseSourceEngine.h"
+#include <glog/logging.h>
+
+#include <cstdio>
+#include <stdexcept>
+#include <string>
+
+#include "../Utils/FileUtils.h"
+
+#ifdef COMPILE_WITH_Ros
+
+namespace InfiniTAM {
+namespace Engine {
+
+RosPoseSourceEngine::RosPoseSourceEngine(ros::NodeHandle& nh)
+    : PoseSourceEngine(),
+      first_time_tf_available_(true),
+      broadcast_transformations(false) {
+  set_camera_pose_ = true;
+
+  // set the topic and frame_id names.
+  nh.param<std::string>("camera_frame_id", camera_frame_id_,
+                        "sr300_depth_optical_frame");
+  nh.param<std::string>("world_frame_id", world_frame_id_, "world");
+  nh.param<std::string>("complete_cloud", complete_cloud_topic_,
+                        "/complete_cloud");
+}
+
+RosPoseSourceEngine::~RosPoseSourceEngine() {
+}
+
+// Get the pose of the camera from the forward kinematics of the robot.
+void RosPoseSourceEngine::TFCallback(const tf::tfMessage& tf_msg) {
+  try {
+    // get transform from world to camera frame.
+    listener.lookupTransform(world_frame_id_, camera_frame_id_, ros::Time(0),
+                             tf_world_to_camera_transform_current_);
+
+    // get the first transformation between the world and the camera.
+    if (first_time_tf_available_) {
+      CHECK_NOTNULL(main_engine_);
+      tf_world_to_camera_transform_at_start_ =
+          tf_world_to_camera_transform_current_;
+      // set the infinitam inertial coordinate system.
+      Vector3f zero_vector;
+      zero_vector.x = zero_vector.y = zero_vector.z = 0;
+      Matrix3f identity;
+      identity.setZeros();
+      identity.m00 = identity.m11 = identity.m22 = 1;
```

Comment:
is there no method to set identity?
if not, 1.0f, but I think there should be one.